app/models/manageiq/providers/azure/cloud_manager.rb

```diff
@@ -73,4 +77,80 @@ def vm_restart(vm, _options = {})
   rescue => err
     _log.error "vm=[#{vm.name}], error: #{err}"
   end
+
+  # Discovery
+
+  # Create EmsAzure instances for all regions with instances
+  # or images for the given authentication.  Created EmsAzure instances
+  # will automatically have EmsRefreshes queued up.  If this is a greenfield
+  # discovery, we will at least add an EmsAzure for eastus
+  def self.discover(clientid, clientkey, azure_tenant_id)
```

Comment:
This looks largely copy-paste from the Amazon code...maybe not for this PR, but we should think about refactoring into some common method.